aws/resource_instance_profile.go

```diff
@@ -138,6 +147,7 @@ func ResourceInstanceProfile() *schema.Resource {
 	instanceProfileSchema := common.StructToSchema(InstanceProfileInfo{},
 		func(m map[string]*schema.Schema) map[string]*schema.Schema {
 			m["instance_profile_arn"].ValidateDiagFunc = ValidInstanceProfile
+			m["iam_role_arn"].ValidateDiagFunc = ValidIamRole
```

Comment:
it should allow empty string as well if it's optional

Reply:
Do you mean the `ValidIamRole` validation function? And allowing empty string as opposed to `null` or missing parameter?

Reply:
when string initialized it's getting initialized to empty string, not to `null`